app/services/cache/MongoCacheClient.scala

```diff
@@ -129,40 +148,45 @@ class MongoCacheClient(appConfig: ApplicationConfig, db: () => DefaultDB, applic
     }
 
     fetchAll(Some(credId)) flatMap { maybeNewCache =>
-
       val cache: Cache = maybeNewCache.getOrElse(Cache(credId, Map.empty))
 
+      val d= cache.data + (key -> jsonData)
+
       val updatedCache: Cache = cache.copy(
         id = credId,
         data = cache.data + (key -> jsonData),
-        lastUpdated = DateTime.now(DateTimeZone.UTC)
+        lastUpdated = LocalDateTime.now(ZoneOffset.UTC)
       )
 
-      val document = Json.toJson(updatedCache)
-      val modifier = BSONDocument("$set" -> document)
-
-      collection.update(ordered = false).one(bsonIdQuery(credId), modifier, upsert = true) map { _ => updatedCache }
+        collection.replaceOne(
+          filter = Filters.equal("_id", credId),
+          replacement = updatedCache,
+          ReplaceOptions().upsert(true)
+        ).toFuture().map { _ => updatedCache}
     }
   }
 
-
   /**
     * Removes the item with the specified key from the cache
     */
   def removeByKey[T](credId: String, key: String): Future[Cache] = {
-
     fetchAll(Some(credId)) flatMap { maybeNewCache =>
       val cache = maybeNewCache.getOrElse(Cache(credId, Map.empty))
 
       val updatedCache = cache.copy(
         data = cache.data - (key),
-        lastUpdated = DateTime.now(DateTimeZone.UTC)
+        lastUpdated = LocalDateTime.now(ZoneOffset.UTC)
```

Comment:
⛏️ This object is also unused.

Reply:
looking into this as it's a little funky

Reply:
Upon a closer look I believe this is used to make the use of cache.copy easier - this is to do with immutable types I believe.

Reply:
I meant: you can delete this `val updatedCache` here and just use `cache.data - key` (you're only using `updatedCache.data` below and not the whole `updatedCache`).

`.copy` creates a copy of an object with new properties assigned. Since you're only using one property below, which you're assigning here, you can skip this step.

```scala
val updatedData = cache.data - (key)
...
Codecs.toBson(updatedData)
// or
...
Codecs.toBson(cache.data - (key))
```

Reply:
Sorry got confused, I actually made that change and then for some reason said that it couldn't be done 🤦 
Maybe not enough caffeine in me... or too much? 🤔